actions/channel_actions.jsx

```diff
@@ -64,36 +64,43 @@ export async function openGroupChannelToUsers(userIds, success, error) {
     }
 }
 
-export async function loadChannelsForCurrentUser() {
-    const state = getState();
-    await ChannelActions.fetchMyChannelsAndMembers(getCurrentTeamId(state))(dispatch, getState);
-    loadDMsAndGMsForUnreads();
-    loadProfilesForSidebar();
+export function loadChannelsForCurrentUser() {
+    return async (dispatch, getState) => {
+        const state = getState();
+
+        await dispatch(ChannelActions.fetchMyChannelsAndMembers(getCurrentTeamId(state)));
+        dispatch(loadDMsAndGMsForUnreads());
+
+        loadProfilesForSidebar();
+    };
 }
 
 export function loadDMsAndGMsForUnreads() {
```

Comment:
I'm not sure if this should be an action as we're not using `dispatch` internally and we're not calling `loadDMsAndGMsForUnreads` anywhere else.  Maybe just combine to `loadChannelsForCurrentUser`?